RecoLocalTracker/SiPixelRecHits/plugins/gpuPixelRecHits.h

```diff
@@ -15,19 +15,17 @@ namespace gpuPixelRecHits {
 
   __global__ void getHits(pixelCPEforGPU::ParamsOnGPU const* __restrict__ cpeParams,
                           BeamSpotCUDA::Data const* __restrict__ bs,
-                          uint16_t const* __restrict__ id,
-                          uint16_t const* __restrict__ x,
-                          uint16_t const* __restrict__ y,
-                          uint16_t const* __restrict__ adc,
+                          SiPixelDigisCUDA::DeviceConstView const * __restrict__ pdigis,
+                          int numElements,
                           uint32_t const* __restrict__ digiModuleStart,
                           uint32_t const* __restrict__ clusInModule,
                           uint32_t const* __restrict__ moduleId,
-                          int32_t const* __restrict__ clus,
-                          int numElements,
                           uint32_t const* __restrict__ hitsModuleStart,
                           TrackingRecHit2DSOAView* phits) {
     auto& hits = *phits;
 
+    auto const digis = *pdigis; // the copy is intentional!
```

Comment:
```auto const & digis``` makes this version 10% slower.
if all ```digis.clus(i) etc ``` were not locally copied it was 20% slower...

Reply:
Could you add more elaborate comments along the first one becoming slow if copied, and the latter one becoming slow if not copied?

Reply:
would I know.... (I have not looked at ptx_
It is only an observation based on nvprof. I suspect TrackingRecHit2DSOAView is large and is really copied "somewhere" (it is also declared non const...)
while SiPixelDigisCUDA::DeviceConstView most probably just goes in registers.

The point is: more investigative work is needed to understand the origin of performance differences: do not exclude that changing compiler (say clang) changes performance diff as well

Reply:
Fine, but I think it would be useful to include these notes also in the code as comments.

Reply:
ok, will do

Reply:
Thanks!

Reply:
Thanks!